### PR TITLE
[s] Creates a gift blacklist. adds chaos ball to it

### PR DESCRIPTION
**Pull request by Super3222**
#### Changelog

:cl:
fix: You can no longer find chaos basketballs in gifts.
/:cl:

also i forgot how to sql, can someone hit me up with a copypasta on how to call a proc on all types of an object

### CODE REVIEW

File: code/game/objects/items/weapons/gift.dm [hunk 1]
```diff
@@ -7,6 +7,7 @@
 /*
  * Gifts
  */
+
 /obj/item/weapon/a_gift
 	name = "gift"
 	desc = "PRESENTS!!!! eek!"
```
Comment (ShadowDeath6) [inline code/game/objects/items/weapons/gift.dm:13]: can you fix the caps here pls

File: code/game/objects/items/weapons/gift.dm [hunk 2]
```diff
@@ -30,6 +31,9 @@
 		M << "<span class='warning'>You're supposed to be spreading gifts, not opening them yourself!</span>"
 		return
 
+	var/static/list/gift_type_blacklist = list()
```
Comment (ShadowDeath6) [inline code/game/objects/items/weapons/gift.dm]: wouldn't this be a whitelist rather than a blacklist

Comment (Super3222) [reply code/game/objects/items/weapons/gift.dm]: i feel blacklist refers to the removal of something. whitelist is when it's attainable, but only excluded until certain occasions

File: code/game/objects/items/weapons/gift.dm [hunk 3]
```diff
@@ -30,6 +31,8 @@
 		M << "<span class='warning'>You're supposed to be spreading gifts, not opening them yourself!</span>"
 		return
 
+	var/static/list/gift_type_blacklist = list(subtypesof(/obj/item/toy/beach_ball/holoball/chaos + (typesof(/obj/item/toy/cards) + /obj/item/toy/cards/deck) - /obj/item/toy/figure - /obj/item/toy/ammo - /obj/item/toy/toy_mech - /obj/item/toy/toy_mech/ranged - /obj/item/toy/toy_mech/melee))
```
Comment (Cruix) [inline code/game/objects/items/weapons/gift.dm]: Flip all the + and - except the first one

**Comment by ShadowDeath6** [comment]
this is actually pretty important to merge asap

**Comment by ShadowDeath6** [comment]
and then the station got nuked by the clerk three times in one day

**Comment by Super3222** [comment]
A merge wave is upon the horizon.

**Comment by ShadowDeath6** [comment]
5 days ago

**Comment by Super3222** [comment]
Just let Cruix get to it when he's able. It's unfair to push him to merge things when he can't update the server at the same time.

**Pull request merged**